### PR TITLE
Test: StakeLimitUtils

### PR DESCRIPTION
**Pull request by TheDZhon**
Tests for `StakeLimitUtils.sol`:
- [x] Make a harness contract
- [x] Add storage lib tests
- [x] Add stake limit lib tests
- [x] Review coverage
- [x] Handle events in tests
- [x] Remove outdated mock `contracts/0.4.24/test_helpers/StakeLimitUtilsMock.sol`

### CODE REVIEW

File: test/0.4.24/lib/stakeLimitUtils.test.ts
Comment (tamtamchik) [inline test/0.4.24/lib/stakeLimitUtils.test.ts:1]: It looks like a cool template for the HardHar test on that step! 🥇

Comment (TheDZhon) [reply test/0.4.24/lib/stakeLimitUtils.test.ts:1]: haha, ok

Comment (TheDZhon) [reply test/0.4.24/lib/stakeLimitUtils.test.ts:1]: would need to discuss do we need templates, hmm

File: contracts/0.4.24/test_helpers/StakeLimitUtilsMock.sol
Comment (tamtamchik) [inline contracts/0.4.24/test_helpers/StakeLimitUtilsMock.sol:1]: 🔥 🙏

**Review by tamtamchik — APPROVED** [review]
:shipit:

**Pull request merged**